{"nbformat":4,"nbformat_minor":0,"metadata":{"colab":{"name":"Copy of networks-Faraz test.ipynb","provenance":[],"toc_visible":true},"kernelspec":{"name":"python3","display_name":"Python 3"}},"cells":[{"cell_type":"markdown","metadata":{"id":"ICtGdgvjs1tY"},"source":["# Networks: AMP-PD PPMI x PDBP - January 2021\n"," - **Project:** Build networks based on underlying data and Mary's feature importance.\n"," - **Author(s):** Mary Makarious, Dan Vitale, Mike Nalls \n","\n","---\n","### Quick Description: \n","- **Problem:** These genetic features are important, are then new networks / communities? \n","- **Solution:** The list below sums it up well ...  \n","\n","1.   Pull genes - this includes querying QTLs from open targets API in brain tissues to connect SNPs to genes.\n","2.   Extract from AMP data.\n","3.   Build correlation / graph space in cases.\n","4.   Leiden to build communties.\n","5.   Vizualize.\n","\n","\n"]},{"cell_type":"markdown","metadata":{"id":"egGsImCBhIS0"},"source":["# Imports and set up."]},{"cell_type":"code","metadata":{"id":"18_B93lgtSPJ","colab":{"base_uri":"https://localhost:8080/"},"executionInfo":{"status":"ok","timestamp":1610302071491,"user_tz":300,"elapsed":36201,"user":{"displayName":"Faraz Faghri","photoUrl":"","userId":"17103565187007973269"}},"outputId":"cf54bba9-e37e-43b5-cdf8-ce18f99ef95d"},"source":["# Imports and set up of the colab.\n","import os\n","from google.colab import drive\n","import pandas.util.testing as tm\n","import h5py\n","import numpy as np\n","import pandas as pd\n","import math\n","import sys\n","import joblib\n","import subprocess\n","import statsmodels.api as sm\n","from scipy import stats\n","\n","! pip install --upgrade tables\n","import tables\n","\n","drive.mount('/content/drive/')\n","os.chdir(\"/content/drive/Shared drives/LNG/users/makariousmb/projects/PPMIxPDBP_GenoML/mike_networks\")\n","! pwd\n","\n","import requests\n","import pandas as pd\n","import json"],"execution_count":1,"outputs":[{"output_type":"stream","text":["/usr/local/lib/python3.6/dist-packages/ipykernel_launcher.py:4: FutureWarning: pandas.util.testing is deprecated. Use the functions in the public API at pandas.testing instead.\n","  after removing the cwd from sys.path.\n"],"name":"stderr"},{"output_type":"stream","text":["Collecting tables\n","\u001b[?25l  Downloading https://files.pythonhosted.org/packages/ed/c3/8fd9e3bb21872f9d69eb93b3014c86479864cca94e625fd03713ccacec80/tables-3.6.1-cp36-cp36m-manylinux1_x86_64.whl (4.3MB)\n","\u001b[K     |████████████████████████████████| 4.3MB 5.9MB/s \n","\u001b[?25hRequirement already satisfied, skipping upgrade: numpy>=1.9.3 in /usr/local/lib/python3.6/dist-packages (from tables) (1.19.4)\n","Requirement already satisfied, skipping upgrade: numexpr>=2.6.2 in /usr/local/lib/python3.6/dist-packages (from tables) (2.7.1)\n","Installing collected packages: tables\n","  Found existing installation: tables 3.4.4\n","    Uninstalling tables-3.4.4:\n","      Successfully uninstalled tables-3.4.4\n","Successfully installed tables-3.6.1\n","Mounted at /content/drive/\n","/content/drive/Shared drives/LNG/users/makariousmb/projects/PPMIxPDBP_GenoML/mike_networks\n"],"name":"stdout"}]},{"cell_type":"markdown","metadata":{"id":"kqB1pCJUhMlQ"},"source":["# Pull the genes of interest from the feature importance file."]},{"cell_type":"markdown","metadata":{"id":"-DEbAc-ChgSF"},"source":["Read in features and extract the 'ENSG' to one list and rsIDs to another."]},{"cell_type":"code","metadata":{"id":"1cQtVG3qhSwP","executionInfo":{"status":"ok","timestamp":1610302073793,"user_tz":300,"elapsed":13824,"user":{"displayName":"Faraz Faghri","photoUrl":"","userId":"17103565187007973269"}}},"source":["# Quick and dirty styles.\r\n","! grep 'ENSG' validation_PDBP.finalHarmonizedCols_toKeep.txt > rna_from_impMat.txt\r\n","! grep 'rs' validation_PDBP.finalHarmonizedCols_toKeep.txt > snps_from_impMat.txt"],"execution_count":2,"outputs":[]},{"cell_type":"code","metadata":{"id":"jH5gMHBFkgpx"},"source":["# API calls set up.\r\n","\r\n","def search_rsid_open_targets_query(rsid):\r\n","\r\n","  \"\"\"\r\n","  query open targets for snp ids in chr_pos_a1_a2 and allele freqs with rsid\r\n","\r\n","  input: rsid\r\n","  output: ids and allele frequencies for different ancestry groups with columns:\r\n","  id\trsId\tgnomadAFR\tgnomadAMR\tgnomadASJ\tgnomadEAS\tgnomadFIN\tgnomadNFE\t\r\n","    gnomadNFEEST\tgnomadNFENWE\tgnomadNFESEU\tgnomadNFEONF\tgnomadOTH\r\n","\r\n","  Some information on the allele frequencies being pulled \r\n","\r\n","  \"gnomAD Allele frequency (African/African-American population)\"\r\n","    gnomadAFR: Float\r\n","\r\n","    \"gnomAD Allele frequency (Latino/Admixed American population)\"\r\n","    gnomadAMR: Float\r\n","\r\n","    \"gnomAD Allele frequency (Ashkenazi Jewish population)\"\r\n","    gnomadASJ: Float\r\n","\r\n","    \"gnomAD Allele frequency (East Asian population)\"\r\n","    gnomadEAS: Float\r\n","\r\n","    \"gnomAD Allele frequency (Finnish population)\"\r\n","    gnomadFIN: Float\r\n","\r\n","    \"gnomAD Allele frequency (Non-Finnish European population)\"\r\n","    gnomadNFE: Float\r\n","\r\n","    \"gnomAD Allele frequency (Non-Finnish Eurpoean Estonian sub-population)\"\r\n","    gnomadNFEEST: Float\r\n","\r\n","    \"gnomAD Allele frequency (Non-Finnish Eurpoean North-Western European sub-population)\"\r\n","    gnomadNFENWE: Float\r\n","\r\n","    \"gnomAD Allele frequency (Non-Finnish Eurpoean Southern European sub-population)\"\r\n","    gnomadNFESEU: Float\r\n","\r\n","    \"gnomAD Allele frequency (Non-Finnish Eurpoean Other non-Finnish European sub-population)\"\r\n","    gnomadNFEONF: Float\r\n","\r\n","    \"gnomAD Allele frequency (Other (population not assigned) population)\"\r\n","    gnomadOTH: Float\r\n","    \"\"\"\r\n","\r\n","  api_query = \"\"\"\r\n","  query search($queryString: String!){\r\n","    search(queryString: $queryString) {\r\n","      variants{\r\n","        id\r\n","        rsId\r\n","        gnomadAFR\r\n","        gnomadAMR\r\n","        gnomadASJ\r\n","        gnomadEAS\r\n","        gnomadFIN\r\n","        gnomadNFE\r\n","        gnomadNFEEST\r\n","        gnomadNFENWE\r\n","        gnomadNFESEU\r\n","        gnomadNFEONF\r\n","        gnomadOTH\r\n","      }\r\n","    }\r\n","  }\"\"\"\r\n","\r\n","  #set base_url for Open Targets Genetics Portal API\r\n","  base_url = \"http://genetics-api.opentargets.io/graphql\"\r\n","  #set variables object\r\n","  variables = {\"queryString\": rsid}\r\n","\r\n","  #perform API call and check status code of response\r\n","  r = requests.post(base_url, json={'query':api_query, \"variables\":variables})\r\n","  if str(r.status_code) == '400':\r\n","    print(f'{rsid} query status code: {r.status_code}')\r\n","  \r\n","  else:\r\n","    pass\r\n","\r\n","  #transform API response into JSON \r\n","  api_response_as_json = json.loads(r.text)\r\n","  \r\n","  return api_response_as_json\r\n","\r\n","def query_qtls(snp_list):\r\n","  \"\"\"\r\n","  uses qtl_coloc_open_targets_query() function to query open targets for qtls\r\n","    given a list of snps in chr_pos_a1_a2 format (hg38)\r\n","\r\n","  input: list of snps (chr_pos_a1_a2 formatted with hg38 positions)\r\n","  output: cleaned up dataframe of qtls with following columns:\r\n","  snp\tgene_symbol\tgene_id\ttype\ttissue\tbeta\tpval\r\n","  \"\"\"\r\n","\r\n","  total_qtls_df = pd.DataFrame()\r\n","  for snp in snp_list:\r\n","    qtl_query = qtl_coloc_open_targets_query(snp)\r\n","    qtl_query_df = pd.json_normalize(qtl_query['data']['genesForVariant'])\r\n","    if len(qtl_query_df) !=  0:\r\n","      qtl_query_df['snp'] = snp\r\n","      qtls_df = qtl_query_df.loc[qtl_query_df.qtls.map(lambda d: len(d)) > 0].reset_index(drop=True).copy()\r\n","      total_qtls_df = total_qtls_df.append(qtls_df)\r\n","  total_qtls_df = total_qtls_df.reset_index(drop=True)\r\n","\r\n","  final_qtls_df = pd.DataFrame()\r\n","  for i, qtl in enumerate(total_qtls_df.qtls):\r\n","    for j, qt in enumerate(qtl):\r\n","      for k, tissue in enumerate(qt['tissues']):\r\n","\r\n","        qtl_dict = {\r\n","            'snp': total_qtls_df.loc[i,'snp'],\r\n","            'gene_symbol': total_qtls_df.loc[i,'gene.symbol'],\r\n","            'gene_id': total_qtls_df.loc[i,'gene.id'],\r\n","            'type': qt['typeId'],\r\n","            'tissue': tissue['tissue']['name'],\r\n","            'beta': tissue['beta'],\r\n","            'pval': tissue['pval']\r\n","        }\r\n","        \r\n","        qtl_dict_df = pd.DataFrame.from_records([qtl_dict])\r\n","        final_qtls_df = final_qtls_df.append(qtl_dict_df).reset_index(drop=True)\r\n","        \r\n","  return final_qtls_df\r\n","\r\n","  \r\n","def qtl_coloc_open_targets_query(variant_id):\r\n","  \"\"\"\r\n","  queries open targets for qtl data given variant_id in chr_pos_a1_a2 format \r\n","    (hg38)\r\n","\r\n","  input: variant id in chr_pos_a1_a2 format (hg38)\r\n","  output: dataframe with the following columns:\r\n","  qtls gene.symbol gene.id\r\n","  \"\"\"\r\n","  api_query = f'''query {{\r\n","    genesForVariant(variantId: \"{variant_id}\") {{\r\n","      gene {{\r\n","        id\r\n","        symbol\r\n","      }}\r\n","      qtls{{\r\n","        typeId\r\n","        sourceId\r\n","        aggregatedScore\r\n","        tissues{{\r\n","          tissue {{\r\n","            id\r\n","            name\r\n","          }}\r\n","          quantile\r\n","          beta\r\n","          pval\r\n","        }}\r\n","      }}\r\n","    }}\r\n","  }}'''\r\n","\r\n","  #set base_url for Open Targets Genetics Portal API\r\n","  base_url = \"http://genetics-api.opentargets.io/graphql\"\r\n","\r\n","  #perform API call and check status code of response\r\n","  r = requests.post(base_url, json={'query': api_query})\r\n","  if str(r.status_code) == '400':\r\n","    print(f'{variant_id} query status code: {r.status_code}')\r\n","  \r\n","  else:\r\n","    pass\r\n","\r\n","  #transform API response into JSON \r\n","  api_response_as_json = json.loads(r.text)\r\n","  \r\n","  return api_response_as_json\r\n","\r\n"],"execution_count":null,"outputs":[]},{"cell_type":"markdown","metadata":{"id":"aFdjSg8MhpkX"},"source":["Query open targets API. Go from rsID to gr38 ID to filtered gene IDs taken from brain QTLs."]},{"cell_type":"code","metadata":{"id":"yJ9hQJXChxe_"},"source":["# Get the variant style IDs as opposed to the rsIDs to query open targets API.\r\n","\r\n","SNP_list_df = pd.read_csv(\"snps_from_impMat.txt\", names=['SNP'])\r\n","SNP_list = SNP_list_df['SNP'].unique()\r\n","\r\n","these_SNPs = []\r\n","\r\n","for i in range(len(SNP_list)):\r\n","  this_SNP = SNP_list[i]\r\n","  rsid_query = search_rsid_open_targets_query(this_SNP)\r\n","  rsid_query_df = pd.json_normalize(rsid_query['data']['search']['variants'])\r\n","  if hasattr(rsid_query_df, 'id'):\r\n","    this_variant_stringed = str(rsid_query_df['id'])\r\n","    this_variant = this_variant_stringed.split()[1]\r\n","  else:\r\n","    this_variant = np.nan\r\n","  this_variant = this_variant_stringed.split()[1]\r\n","  print(this_SNP, this_variant)\r\n","  these_SNPs.append((this_SNP, this_variant))\r\n","\r\n","these_SNPs_df = pd.DataFrame(these_SNPs, columns=('SNP', 'VARIANT'))\r\n","these_SNPs_df.to_csv(\"variants_mined_01072021.csv\", index=False)\r\n"],"execution_count":null,"outputs":[]},{"cell_type":"code","metadata":{"id":"v75maXllGP_b"},"source":["# Now get the gene IDs for QTLs in brain tissues relating to these SNPs.\r\n","\r\n","these_SNPs_df = pd.read_csv(\"variants_mined_01072021.csv\")\r\n","variant_list = these_SNPs_df['VARIANT'].unique()\r\n","\r\n","qtls_df = query_qtls(variant_list)\r\n","qtls_df.head()\r\n","\r\n","qtls_df.to_csv(\"QTLS_mined_01072021.csv\", index=False)\r\n"],"execution_count":null,"outputs":[]},{"cell_type":"code","metadata":{"id":"Lu4ydNu7s3Pw"},"source":["# Now subset that QTL list to tissues of interest."],"execution_count":null,"outputs":[]},{"cell_type":"markdown","metadata":{"id":"leIK6Tyul7b_"},"source":["Merge the two gene lists and make unique."]},{"cell_type":"code","metadata":{"id":"R5BqyvBumAH7"},"source":["qtls_df = pd.read_csv(\"QTLS_mined_01072021.csv\", engine='c')\r\n","rna_df = pd.read_csv(\"rna_from_impMat.txt\", engine='c', names=['gene_id'])"],"execution_count":null,"outputs":[]},{"cell_type":"code","metadata":{"colab":{"base_uri":"https://localhost:8080/","height":204},"id":"t4eSaIdl8SyQ","executionInfo":{"status":"ok","timestamp":1610293041590,"user_tz":300,"elapsed":285,"user":{"displayName":"Mike Nalls","photoUrl":"https://lh3.googleusercontent.com/a-/AOh14GjDjNx7RfkDi5NwxyPHo2_ijZ7TyI1irNRya74-2g=s64","userId":"13912922682160491421"}},"outputId":"409ac06c-40e0-4995-c0a5-e27d13351ce4"},"source":["qtls_df.head()\r\n","# qtls_df.describe()"],"execution_count":null,"outputs":[{"output_type":"execute_result","data":{"text/html":["<div>\n","<style scoped>\n","    .dataframe tbody tr th:only-of-type {\n","        vertical-align: middle;\n","    }\n","\n","    .dataframe tbody tr th {\n","        vertical-align: top;\n","    }\n","\n","    .dataframe thead th {\n","        text-align: right;\n","    }\n","</style>\n","<table border=\"1\" class=\"dataframe\">\n","  <thead>\n","    <tr style=\"text-align: right;\">\n","      <th></th>\n","      <th>snp</th>\n","      <th>gene_symbol</th>\n","      <th>gene_id</th>\n","      <th>type</th>\n","      <th>tissue</th>\n","      <th>beta</th>\n","      <th>pval</th>\n","    </tr>\n","  </thead>\n","  <tbody>\n","    <tr>\n","      <th>0</th>\n","      <td>4_47509764_T_C</td>\n","      <td>ATP10D</td>\n","      <td>ENSG00000145246</td>\n","      <td>eqtl</td>\n","      <td>Lcl (TWINSUK)</td>\n","      <td>0.090071</td>\n","      <td>7.271000e-07</td>\n","    </tr>\n","    <tr>\n","      <th>1</th>\n","      <td>4_47509764_T_C</td>\n","      <td>ATP10D</td>\n","      <td>ENSG00000145246</td>\n","      <td>eqtl</td>\n","      <td>Lcl (GENCORD)</td>\n","      <td>0.154816</td>\n","      <td>5.169660e-06</td>\n","    </tr>\n","    <tr>\n","      <th>2</th>\n","      <td>4_47509764_T_C</td>\n","      <td>NFXL1</td>\n","      <td>ENSG00000170448</td>\n","      <td>eqtl</td>\n","      <td>Blood (eQTLGen)</td>\n","      <td>0.076645</td>\n","      <td>1.287900e-19</td>\n","    </tr>\n","    <tr>\n","      <th>3</th>\n","      <td>18_49054943_G_A</td>\n","      <td>DYM</td>\n","      <td>ENSG00000141627</td>\n","      <td>eqtl</td>\n","      <td>Blood (eQTLGen)</td>\n","      <td>0.165194</td>\n","      <td>9.361700e-79</td>\n","    </tr>\n","    <tr>\n","      <th>4</th>\n","      <td>4_25814426_A_G</td>\n","      <td>SEL1L3</td>\n","      <td>ENSG00000091490</td>\n","      <td>eqtl</td>\n","      <td>Monocyte naive (FAIRFAX 2014)</td>\n","      <td>-0.409197</td>\n","      <td>4.979690e-06</td>\n","    </tr>\n","  </tbody>\n","</table>\n","</div>"],"text/plain":["               snp gene_symbol  ...      beta          pval\n","0   4_47509764_T_C      ATP10D  ...  0.090071  7.271000e-07\n","1   4_47509764_T_C      ATP10D  ...  0.154816  5.169660e-06\n","2   4_47509764_T_C       NFXL1  ...  0.076645  1.287900e-19\n","3  18_49054943_G_A         DYM  ...  0.165194  9.361700e-79\n","4   4_25814426_A_G      SEL1L3  ... -0.409197  4.979690e-06\n","\n","[5 rows x 7 columns]"]},"metadata":{"tags":[]},"execution_count":4}]},{"cell_type":"code","metadata":{"colab":{"base_uri":"https://localhost:8080/","height":204},"id":"_ZKJypuH8W9C","executionInfo":{"status":"ok","timestamp":1610293074916,"user_tz":300,"elapsed":322,"user":{"displayName":"Mike Nalls","photoUrl":"https://lh3.googleusercontent.com/a-/AOh14GjDjNx7RfkDi5NwxyPHo2_ijZ7TyI1irNRya74-2g=s64","userId":"13912922682160491421"}},"outputId":"a0fa332e-4c7a-4313-fcae-862663d588c0"},"source":["rna_df.head()\r\n","# rna_df.describe()"],"execution_count":null,"outputs":[{"output_type":"execute_result","data":{"text/html":["<div>\n","<style scoped>\n","    .dataframe tbody tr th:only-of-type {\n","        vertical-align: middle;\n","    }\n","\n","    .dataframe tbody tr th {\n","        vertical-align: top;\n","    }\n","\n","    .dataframe thead th {\n","        text-align: right;\n","    }\n","</style>\n","<table border=\"1\" class=\"dataframe\">\n","  <thead>\n","    <tr style=\"text-align: right;\">\n","      <th></th>\n","      <th>gene_id</th>\n","    </tr>\n","  </thead>\n","  <tbody>\n","    <tr>\n","      <th>0</th>\n","      <td>ENSG00000000938</td>\n","    </tr>\n","    <tr>\n","      <th>1</th>\n","      <td>ENSG00000002919</td>\n","    </tr>\n","    <tr>\n","      <th>2</th>\n","      <td>ENSG00000003509</td>\n","    </tr>\n","    <tr>\n","      <th>3</th>\n","      <td>ENSG00000004478</td>\n","    </tr>\n","    <tr>\n","      <th>4</th>\n","      <td>ENSG00000004948</td>\n","    </tr>\n","  </tbody>\n","</table>\n","</div>"],"text/plain":["           gene_id\n","0  ENSG00000000938\n","1  ENSG00000002919\n","2  ENSG00000003509\n","3  ENSG00000004478\n","4  ENSG00000004948"]},"metadata":{"tags":[]},"execution_count":8}]},{"cell_type":"code","metadata":{"colab":{"base_uri":"https://localhost:8080/"},"id":"TV_wECTB8n9X","executionInfo":{"status":"ok","timestamp":1610294096559,"user_tz":300,"elapsed":274,"user":{"displayName":"Mike Nalls","photoUrl":"https://lh3.googleusercontent.com/a-/AOh14GjDjNx7RfkDi5NwxyPHo2_ijZ7TyI1irNRya74-2g=s64","userId":"13912922682160491421"}},"outputId":"7c3d148c-75c1-4473-b3a5-2c710ccc411b"},"source":["qtl_gene_ids = qtls_df['gene_id'].unique()\r\n","rna_gene_ids = rna_df['gene_id'].unique()\r\n","gene_ids_to_pull = np.concatenate([qtl_gene_ids, rna_gene_ids])\r\n","unique_gene_ids_to_pull = pd.unique(gene_ids_to_pull)"],"execution_count":null,"outputs":[{"output_type":"execute_result","data":{"text/plain":["9797"]},"metadata":{"tags":[]},"execution_count":21}]},{"cell_type":"markdown","metadata":{"id":"fWDhcnLBhzJG"},"source":["Pull these genes from gene expression data."]},{"cell_type":"code","metadata":{"id":"pdGDDBCII_sf"},"source":["# Load in and reduce the working data.\n","raw_df = pd.read_hdf(\"../mike_diff_exp/data_for_diff_exp.h5\", key='diff_exp_df', mode='r')\n","case_df = raw_df[raw_df['PHENO'] == 1]\n","data_df = case_df[unique_gene_ids_to_pull]"],"execution_count":null,"outputs":[]},{"cell_type":"markdown","metadata":{"id":"pE7tdf0hhTka"},"source":["# Make the correlation matrix / graph space in cases only.\r\n","Then filter it at 0.8 positve correlation."]},{"cell_type":"code","metadata":{"id":"P31EdmguFBer"},"source":["# Quick imports.\r\n","import networkx as nx\r\n","import community\r\n","from cdlib import algorithms, readwrite\r\n","# ! pip install cdlib"],"execution_count":null,"outputs":[]},{"cell_type":"code","metadata":{"id":"iMztHM0zh9lZ"},"source":["# Correlate the variables.\r\n","cor_matrix = data_df.iloc[:,1:].corr()\r\n","links = cor_matrix.stack().reset_index()\r\n","links.columns = ['feature1', 'feature2','corr']"],"execution_count":null,"outputs":[]},{"cell_type":"code","metadata":{"id":"k6a9X0s6Ekp3"},"source":["# filter at positve correlation > 0.8.\r\n","links_filtered = links.loc[ (links['corr'] >= 0.8) & (links['feature1'] != links['feature2']) ]"],"execution_count":null,"outputs":[]},{"cell_type":"code","metadata":{"id":"c2lJvgE7E9Z1"},"source":["# Make the networkx graph.\r\n","G=nx.from_pandas_edgelist(links_filtered, 'feature1', 'feature2')"],"execution_count":null,"outputs":[]},{"cell_type":"markdown","metadata":{"id":"YpSP0JUGh9_R"},"source":["# Leiden communities."]},{"cell_type":"code","metadata":{"id":"qmg9X8GYiAk-"},"source":["coms = algorithms.leiden(G)\r\n","readwrite.write_community_csv(coms, \"test_coms.csv\")"],"execution_count":null,"outputs":[]},{"cell_type":"code","metadata":{"colab":{"base_uri":"https://localhost:8080/","height":282},"id":"B_olZ3qHbX3N","executionInfo":{"status":"error","timestamp":1610301706126,"user_tz":300,"elapsed":324,"user":{"displayName":"Mike Nalls","photoUrl":"https://lh3.googleusercontent.com/a-/AOh14GjDjNx7RfkDi5NwxyPHo2_ijZ7TyI1irNRya74-2g=s64","userId":"13912922682160491421"}},"outputId":"2a323efe-66b1-495f-a409-54430905f52b"},"source":["coms_df = pd.DataFrame.from_dict(coms, orient='index')"],"execution_count":null,"outputs":[{"output_type":"error","ename":"TypeError","evalue":"ignored","traceback":["\u001b[0;31m---------------------------------------------------------------------------\u001b[0m","\u001b[0;31mTypeError\u001b[0m                                 Traceback (most recent call last)","\u001b[0;32m<ipython-input-79-69a8aeb95381>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m()\u001b[0m\n\u001b[0;32m----> 1\u001b[0;31m \u001b[0mcoms_df\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mpd\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mDataFrame\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mfrom_dict\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mcoms\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0morient\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;34m'index'\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m","\u001b[0;32m/usr/local/lib/python3.6/dist-packages/pandas/core/frame.py\u001b[0m in \u001b[0;36mfrom_dict\u001b[0;34m(cls, data, orient, dtype, columns)\u001b[0m\n\u001b[1;32m   1295\u001b[0m         \u001b[0morient\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0morient\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mlower\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   1296\u001b[0m         \u001b[0;32mif\u001b[0m \u001b[0morient\u001b[0m \u001b[0;34m==\u001b[0m \u001b[0;34m\"index\"\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m-> 1297\u001b[0;31m             \u001b[0;32mif\u001b[0m \u001b[0mlen\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mdata\u001b[0m\u001b[0;34m)\u001b[0m \u001b[0;34m>\u001b[0m \u001b[0;36m0\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m   1298\u001b[0m                 \u001b[0;31m# TODO speed up Series case\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   1299\u001b[0m                 \u001b[0;32mif\u001b[0m \u001b[0misinstance\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mlist\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mdata\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mvalues\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;36m0\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m(\u001b[0m\u001b[0mSeries\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mdict\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n","\u001b[0;31mTypeError\u001b[0m: object of type 'NodeClustering' has no len()"]}]},{"cell_type":"markdown","metadata":{"id":"9JuxMXgdTbqJ"},"source":["Quick page ranking."]},{"cell_type":"code","metadata":{"colab":{"base_uri":"https://localhost:8080/","height":214},"id":"UN9ZgeIYTeH7","executionInfo":{"status":"error","timestamp":1610300572671,"user_tz":300,"elapsed":50581,"user":{"displayName":"Mike Nalls","photoUrl":"https://lh3.googleusercontent.com/a-/AOh14GjDjNx7RfkDi5NwxyPHo2_ijZ7TyI1irNRya74-2g=s64","userId":"13912922682160491421"}},"outputId":"7ecd2af1-cd3c-4905-f9fe-89b18ed1e2a5"},"source":["pageranked = nx.pagerank(G)\r\n","pageranked_df = pd.DataFrame.from_dict(pageranked, orient='index')\r\n","pageranked_df.reset_index(level=0, inplace=True)\r\n","pageranked_df.rename(columns={\"index\":\"feature_name\",0:\"page_rank\"}, inplace=True)\r\n","pageranked_outfile = 'network_pageranks.csv'\r\n","pageranked_df.to_csv(partition_outfile, index=False)\r\n","pageranked_df.head()\r\n"],"execution_count":null,"outputs":[{"output_type":"error","ename":"NameError","evalue":"ignored","traceback":["\u001b[0;31m---------------------------------------------------------------------------\u001b[0m","\u001b[0;31mNameError\u001b[0m                                 Traceback (most recent call last)","\u001b[0;32m<ipython-input-59-7a7a710a9e20>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m()\u001b[0m\n\u001b[1;32m      4\u001b[0m \u001b[0mpageranked_df\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mrename\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mcolumns\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;34m{\u001b[0m\u001b[0;34m\"index\"\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\"feature_name\"\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;36m0\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\"page_rank\"\u001b[0m\u001b[0;34m}\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0minplace\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;32mTrue\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m      5\u001b[0m \u001b[0mpageranked_outfile\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0;34m'network_pageranks.csv'\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m----> 6\u001b[0;31m \u001b[0mpageranked_df\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mto_csv\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mpartition_outfile\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mindex\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;32mFalse\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m      7\u001b[0m \u001b[0mpageranked_df\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mhead\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n","\u001b[0;31mNameError\u001b[0m: name 'partition_outfile' is not defined"]}]},{"cell_type":"markdown","metadata":{"id":"DqWrNvI7iDeZ"},"source":["# A pretty picture to visualize and export the communities."]},{"cell_type":"code","metadata":{"id":"RTbCsxbTiMX-"},"source":[""],"execution_count":null,"outputs":[]}]}